modules/nunjucks_macros.py

```diff
@@ -0,0 +1,62 @@
+"""
+"""
+
+from __future__ import absolute_import
+from docutils import nodes
+from docutils.parsers.rst import Directive, directives
+
+
+class CallMacro(Directive):
+    has_content = True
+    required_arguments = 1
+    optional_arguments = 10
+    final_argument_whitespace = False
+    option_spec = {
+        # 'post': directives.unchanged,
+    }
+
+    def get_macro(self):
+        macro_id = self.arguments[0]
+        args = self.arguments[1:]
+        content = '\n'.join(self.content)
+        try:
+            filename, macro_name = macro_id.split("#")
+        except ValueError:
+            # We want to raise a useful error message:
+            raise ValueError("Invalid macro id: {}".format(macro_id))
+
+        if content:
+            call = """
+            {{% call to_run.{name}({arglist}) %}}
+              {content}
+            {{% endcall %}}
+            """.strip()
+        else:
+            call = "{{{{ to_run.{name}({arglist}) }}}}"
```

Comment:
Yo dawg, I wrote a template for your template so you can render while you render.